tests/nmea_parser.cc

```diff
@@ -0,0 +1,65 @@
+#include "common.h"
+
+// BoB robotics includes
+#include "gps/nmea_parser.h"
+
+// Standard C includes
+#include <cstring>
+
+using namespace BoBRobotics::GPS;
+using namespace units::angle;
+using namespace units::length;
+
+
+TEST(NMEAParser, SampleGNGGASentence)
+{
+    NMEAParser nmea;
+    GPSData data;
+
+    const auto sentence = "$GNGGA,001043.00,4404.14036,N,12118.85961,W,"
+                          "1,12,0.98,1113.0,M,-21.3,M*47";
+    ASSERT_TRUE(nmea.parseCoordinates(sentence, data));
+
+    // Check timestamp
+    EXPECT_EQ(data.time.tm_hour, 0);
+    EXPECT_EQ(data.time.tm_min, 10);
+    EXPECT_EQ(data.time.tm_sec, 43);
+    EXPECT_EQ(data.milliseconds, 0);
+
+    // Check coordinates
+    BOB_EXPECT_UNIT_T_EQ(data.coordinate.lat, degree_t{ 44 } + arcminute_t{ 4.14036 });
+    BOB_EXPECT_UNIT_T_EQ(data.coordinate.lon, degree_t{ 121 } + arcminute_t{ 18.85961 });
+    BOB_EXPECT_UNIT_T_EQ(data.coordinate.height, meter_t{ 1113 });
+
+    // Check self-reported accuracy
+    BOB_EXPECT_UNIT_T_EQ(data.horizontalDilution, meter_t{ 0.98 });
+
+    // Misc
+    EXPECT_EQ(data.gpsQuality, GPSQuality::GPSFIX);
+    EXPECT_EQ(data.numberOfSatellites, 12);
+}
```

Comment:
some 'death' tests of the form:
```c++
 try {
        // **TODO** something
        FAIL();
    }
    catch(const NMEAError &) {
    }
```
would be nice to check the error handling works properly

Reply:
Good idea